torch/distributed/_tensor/api.py

```diff
@@ -107,6 +107,15 @@ def backward(ctx, grad_output: torch.Tensor):  # type: ignore[override]
             None,
         )
 
+def allgather_metadata(local_tensor):
```

Comment:
Let's move this util to `_collective_utils.py` and rename this as `check_tensor_meta`, we can also put the metadata check inside this function and only return a bool that indicates the metadata across ranks are consistent or not.